students/Briadean/lesson05/Mailroom3.py

```diff
@@ -0,0 +1,160 @@
+               "Gori Tsuruta": [25.00, 30.00, 60.00, 90.00],
+               "Washimi Koiwasaki": [10.00, 10.00, 10.00, 35.00, 15.00]}
+
+
+def main_menu_choice():
+    """Display main menu and collect user input"""
+    user_choice = input("\n1: Send a Thank You\n"
+                        "2: Generate a Report\n"
+                        "3: Send Letters to Everyone\n"
+                        "4: Exit the Mailroom\n"
+                        "Please enter your selection from the options above >> ")
+    return user_choice
+
+
+def thank_you_menu():
+    """Display Thank You menu and collect user input"""
+    user_choice = input("\nPlease enter a donor name, "
+                        "'List' to see a list of previous donors, "
+                        "or 'Menu' to return to the menu. >> ").title()
+    while True:
+        if user_choice == "List":
+            print()
+            print(list_donors())
+            # Is there a better way to get back to user_choice = input? Break/return/continue all go back to main menu
+            user_choice = input("\nPlease enter a donor name, "
+                                "'List' to see a list of previous donors, "
+                                "or 'Menu' to return to the menu. >> ").title()
+        elif user_choice == "Menu":
+            return
+        else:
+            accept_donation(user_choice)
+            return
+
+
+def list_donors():
+    """Make a list of donors that's easily displayable"""
+    print("Donor List:")
+    # Why is this printing None at the end of the list?
+    [print(donor) for donor in donors_dict.keys()]
+
+
+def accept_donation(name):
+    """Accept a new donation. If donor does not already exist, add to dictionary
+    and create appropriate data container.
+    Params: input = donor name
+            returns a thank you letter
+    """
+    donors_dict.setdefault(name, [])
+    donors_dict.setdefault(name, [])
+    donation_amt = input("How much are they donating today? >> $ ")
+    donors_dict[name].append(float(donation_amt))
+    print(generate_thank_you(name, float(donation_amt)))
+
+
+def generate_thank_you(name, donation):
+    """Generates a standardized thank you letter
+    Params: input = donor name, most recent donation amounts
+            returns a write_letters_disk
+    """
+    letter = ("\nDear {},\n\n"
+              "Thank you for your recent generous donation of ${:,.2f}.\n"
+              "It will be put to good use.\n\n"
+              "Sincerely,\n"
+              "The Resistance Against Giant Flying Meatballs\n".format(name, donation))
+    return letter
+
+
+def write_letters_disk():
+    """Writes a thank you letter for each donor in the database to disk"""
+    for donor, donation in donors_dict.items():
+        letter = generate_thank_you(donor, donation[-1])
+        filename = donor + ".txt"
+        print("Writing letter for:", donor)
+        open(filename, "w").write(letter)
```

Comment:
Don't forget to close your files! This can be trouble later. I always just use the "with open" so it closes automatically:
```
with open('output.txt', 'w') as f:
    for i in range(10):
       f.write("this is line: %i\n"%i)

Stolen from:
https://uwpce-pythoncert.github.io/PythonCertDevel/modules/Files.html#files
```